src/background/services/storage/schemaMigrations/migrations/permissions_v2.ts

```diff
@@ -0,0 +1,85 @@
+import Joi from 'joi';
+import { rpcErrors } from '@metamask/rpc-errors';
+
+import { CommonError } from '@src/utils/errors';
+
+const VERSION = 2 as const;
+
+type LegacySchema = Record<
+  string,
+  {
+    domain: string;
+    accounts: {
+      [addressC: string]: boolean;
+    };
+  }
+>;
+
+const previousSchema = Joi.object<LegacySchema>({
+  version: Joi.number().valid(1),
+}).pattern(
+  Joi.string().not('version'),
+  Joi.object({
+    domain: Joi.string().required(),
+    accounts: Joi.object().pattern(Joi.string(), Joi.boolean().required()),
+  }),
+);
+
+type NewSchema = {
+  permissions: Record<
+    string,
+    {
+      domain: string;
+      accounts: {
+        [address: string]: 'EVM'; // At the moment of writing this migration, we only stored permissions for EVM addresses
+      };
+    }
+  >;
+  version: typeof VERSION;
+};
+
+const up = (legacyPermissions) => {
+  const validationResult = previousSchema.validate(legacyPermissions);
+
+  if (validationResult.error) {
+    throw rpcErrors.internal({
+      data: {
+        reason: CommonError.MigrationFailed,
+        context: validationResult.error.message,
+      },
+    });
+  }
+
+  const newPermissions = Object.entries(validationResult.value).reduce(
+    (permAcc, [domain, { domain: _, accounts }]) => {
```

Comment:
Why to destructure the domain in the second element? 🤔 
As far as I can tell it is not used anywhere
\#justAsking

Reply:
True 👍